server/build_event_protocol/build_event_server/build_event_server.go

```diff
@@ -7,19 +7,90 @@ import (
 
 	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_asset/fetch_server"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_asset/push_server"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/action_cache_server"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/byte_stream_server"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/capabilities_server"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/content_addressable_storage_server"
 	"github.com/buildbuddy-io/buildbuddy/server/util/log"
 	"github.com/buildbuddy-io/buildbuddy/server/util/status"
 	"github.com/golang/protobuf/ptypes/empty"
 	"google.golang.org/grpc"
 
 	bepb "github.com/buildbuddy-io/buildbuddy/proto/build_events"
 	pepb "github.com/buildbuddy-io/buildbuddy/proto/publish_build_event"
+	rapb "github.com/buildbuddy-io/buildbuddy/proto/remote_asset"
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	scpb "github.com/buildbuddy-io/buildbuddy/proto/scheduler"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
 )
 
 type BuildEventProtocolServer struct {
 	env environment.Env
 }
 
+func Register(env environment.Env, grpcServer *grpc.Server) error {
+	// Register to handle build event protocol messages.
+	buildEventServer, err := NewBuildEventProtocolServer(env)
+	if err != nil {
+		return status.InternalErrorf("Error initializing BuildEventProtocolServer: %s", err)
+	}
+	pepb.RegisterPublishBuildEventServer(grpcServer, buildEventServer)
+
+	enableCache := env.GetCache() != nil
+	// OPTIONAL CACHE API -- only enable if configured.
+	if enableCache {
```

Comment:
I think most of these services (apart from the BEP service itself) are out of place in build_event_server.go
They're part of the overall system but are not part of the BEP server.

Reply:
Moved!